assigner/tests/config_test.py

```diff
@@ -0,0 +1,92 @@
+from assigner.tests.utils import AssignerTestCase
+
+from assigner.config.versions import validate, get_version, upgrade, ValidationError
+from assigner.config.upgrades import UPGRADES
+from assigner.config.schemas import SCHEMAS
+
+
+CONFIGS = [
+    { # Version 0
+        "token": "xxx gitlab token xxx",
+        "gitlab-host": "https://git.gitlab.com",
+        "namespace": "assigner-testing",
+        "semester": "2016-SP",
+        "roster": [],
+    },
+    { # Version 1
+        "gitlab-token": "xxx gitlab token xxx",
+        "gitlab-host": "https://git.gitlab.com",
+        "namespace": "assigner-testing",
+        "semester": "2016-SP",
+        "roster": [],
+    },
+    { # Version 2
+        "version": 2,
+        "backend": {
```

Comment:
@brhoades What do you think of this configuration schema? If you like it, I'll update the rest of the code to work with it.